src/Data.php

```diff
@@ -0,0 +1,220 @@
+    {
+        if (!empty($data) && (is_int($data) || ctype_digit($data))) {
+            return true;
+        }
+
+        return false;
+    }
+
+     /**
+      * Tests if data is a valid float
+      *
+      *     // Returns true
+      *     Data::isFloat(987.65);
+      *
+      *     // Returns false
+      *     Data::isFloat(98765);
+      *
+      * @param   float   $data  data to check
+      * @return  boolean
+      */
+    public static function isFloat($data)
+    {
+        if (!empty($data) && is_float($data)) {
+            return true;
+        }
+
+        return false;
+    }
+   
+     /**
+      * Tests if data is a valid binary string
+      *
+      *     // Returns true
+      *     Data::isBinary('01001000 01100101 01101100 01101100 01101111 00100000 01010111 01101111 01110010 01101100 01100100');
+      *
+      *     // Returns false
+      *     Data::isBinary('some non binary string');
+      *
+      * @param   string   $data  data to check
+      * @return  boolean
+      */
+    public static function isBinary($data)
+    {
+        if (!empty($data) && preg_match(self::BINARY_REGEXP, $data)) {
+            return true;
+        }
+
+        return false;
+    }
+
+     /**
+      * Tests if data is a valid timestamp
+      *
+      *     // Returns true
+      *     Data::isTimestamp('2014-10-31 12:22:36');
+      *
+      *     // Returns false
+      *     Data::isTimestamp('0000-00-00 00:00:00');
+      *
+      * @param   timestamp   $data  data to check
+      * @return  boolean
+      */
+    //TODO: Add support for "real" datetime check, instead of possibly allowing something like 2014-99-99 88:77:66
+    public static function isTimestamp($data)
+    {
+        if (!empty($data) && preg_match(self::TIMESTAMP_REGEXP, $data) && !in_array($data, static::$invalid_timestamps)) {
+            return true;
+        }
+
+        return false;
+    }
+
+     /**
+      * Tests if data is a valid date
+      *
+      *     // Returns true
+      *     Data::isDate('2014-10-31');
+      *
+      *     // Returns false
+      *     Data::isDate('0000-00-00');
+      *
+      * @param   string   $data  data to check
+      * @return  boolean
+      */
+    //TODO: Add support for "real" date check, instead of possibly allowing something like 2014-99-99
+    public static function isDate($data)
+    {
+        if (!empty($data) && preg_match(self::DATE_REGEXP, $data) && !in_array($data, static::$invalid_dates)) {
+            return true;
+        }
+
+        return false;
+    }
+
+     /**
+      * Tests if data is a valid email
+      *
+      *     // Returns true
+      *     Data::isEmail('foo@bar.com');
+      *
+      *     // Returns false
+      *     Data::isEmail('foo[at]bar[dot]com');
+      *
+      * @param   string   $data  data to check
+      * @return  boolean
+      */
+    public static function isEmail($data)
+    {
+        if (filter_var($data, FILTER_VALIDATE_EMAIL)) {
```

Comment:
I think we should leave it as is, for the following reason. I would expect the following code to work:

if( is_email('user@domain.com') === true ){
    //process success code
}

just to stay true to the is_* returning a boolean.